drgn_tools/task.py

```diff
@@ -185,6 +195,173 @@ def get_command(task: Object) -> str:
+    MM_ANONPAGES = prog.constant("MM_ANONPAGES").value_()
+    try:
+        MM_SHMEMPAGES = prog.constant("MM_SHMEMPAGES").value_()
+    except LookupError:
+        MM_SHMEMPAGES = -1
+
+    try:
+        filerss = task.mm.rss_stat.count[MM_FILEPAGES].counter.value_()
+        anonrss = task.mm.rss_stat.count[MM_ANONPAGES].counter.value_()
+        if MM_SHMEMPAGES > 0:
+            shmrss = task.mm.rss_stat.count[MM_SHMEMPAGES].counter.value_()
+        else:
+            shmrss = 0
+
+        rss_mm = filerss + anonrss + shmrss
+
+    except drgn.FaultError:
+        rss_mm = 0
+
+    gid = task.tgid.value_()
+
+    rss_task = rss_task_dict.get(gid)
+
+    if gid in rss_task_dict:
+        rss_task = rss_task_dict[gid]
+    else:
+        rss_task = 0
+        ltask = task.group_leader
+
+        task_filerss = ltask.rss_stat.count[MM_FILEPAGES].value_()
+        task_anonrss = ltask.rss_stat.count[MM_ANONPAGES].value_()
+        if MM_SHMEMPAGES > 0:
+            task_shmrss = ltask.rss_stat.count[MM_SHMEMPAGES].value_()
+        else:
+            task_shmrss = 0
+
+        rss_task += task_filerss + task_anonrss + task_shmrss
+
+        for gtask in list_for_each_entry(
+            "struct task_struct",
+            task.group_leader.thread_group.address_of_(),
+            "thread_group",
+        ):
+            task_filerss = gtask.rss_stat.count[MM_FILEPAGES].value_()
+            task_anonrss = gtask.rss_stat.count[MM_ANONPAGES].value_()
+            if MM_SHMEMPAGES > 0:
+                task_shmrss = gtask.rss_stat.count[MM_SHMEMPAGES].value_()
+            else:
+                task_shmrss = 0
+
+            rss_task += task_filerss + task_anonrss + task_shmrss
+
+        rss_task_dict[gid] = rss_task
+
+    rss = rss_mm + rss_task
+    page_size = prog["PAGE_SIZE"].value_()
+    rss_kb = (rss * page_size) // ByteToKB
+
+    return rss_kb
+
+
+def get_pct_mem(prog: Program, task: Object) -> str:
+    """
+    :returns %MEM value
+    """
+    gid = task.tgid.value_()
+    rss_task = rss_task_dict_global[gid]
+    rss_mm = rss_mm_dict_global[task.pid.value_()]
+    tot_rss = rss_mm + rss_task
+
+    page_shift = prog["PAGE_SHIFT"].value_()
+    high_mem = prog["high_memory"].value_()
+
+    if prog.platform.arch == Architecture.X86_64:
+        try:
+            page_offset_base = prog["page_offset_base"].value_()
+        except KeyError:
+            # Prior to 021182e52fe0 ("x86/mm: Enable KASLR for physical mapping
+            # memory regions"), page_offset_base did not exist and there was
+            # just a constant for the beginning of the memory map.
+            page_offset_base = 0xFFFF880000000000
+
+        total_pages = (high_mem - page_offset_base) >> page_shift
+    else:
+        raise NotImplementedError(
+            "%MEM calculation is not (yet) implemented for aarch64"
+        )
+    mem = (tot_rss * 100) / total_pages
+    pct_mem = "%.1f" % mem
+
+    return pct_mem
```

Comment:
Please return the value as a `float`, and let the caller decide how to format it. It doesn't make sense for `get_pct_mem()` to return a string.